src/Payments/Integrations/Stripe.php

```diff
@@ -49,6 +51,7 @@ final class Stripe extends AbstractPaymentMethodType {
 	 */
 	public function __construct( Api $asset_api ) {
 		$this->asset_api = $asset_api;
+		add_action( 'woocommerce_rest_checkout_process_payment_with_context', [ $this, 'add_payment_request_order_meta' ], 8, 2 );
```

Comment:
Hey this is cheating :D I think this is ok for the beta this week but we should probably use this hook correctly in the integration before "public" launch and avoid the legacy handler.

Reply:
Hmm... what about this is cheating? I'm not sure what the _correct_ use of this hook should be?

Reply:
Referring to `add_payment_request_order_meta` and this which appears to be doing something similar to legacy handling (updating POST and doing it's own legacy handling).

`woocommerce_rest_checkout_process_payment_with_context` was intended to be used to handle the payment itself.

Reply:
Oh gotcha, I saw that `Library::process_legacy_payment` is currently what is handling server side payments right now (and is using the same filter) so I was puzzled by why what I was doing is considered "cheating" :).  I guess ultimately, `Library::process_legacy_payment` implementation is also cheating.

So next steps would be to work out what needs to happen for payment methods in core and payment methods in extensions to remove reliance on the $_POST globals and be able to just hook into the `woocommerce_rest_checkout_process_payment_with_context` hook (as well as still support shortcode based checkout)?

Also, to clarify, the major problem we have here is the manipulation of `$_POST` right? Assuming we got rid of that dependency in the source (extensions and core) code, it's not really utilizing legacy handling right?

Reply:
> So next steps would be to work out what needs to happen for payment methods in core and payment methods in extensions to remove reliance on the $_POST globals and be able to just hook into the woocommerce_rest_checkout_process_payment_with_context hook (as well as still support shortcode based checkout)?

Exactly. What we document needs to do this.

> Also, to clarify, the major problem we have here is the manipulation of $_POST right?

Yes, we really should be avoiding that going forward, passing only the data that the gateways need. That also makes it more flexible in future if we need to take payments in other contexts.